### PR TITLE
Add Proguard Rules for Graal annotations

### LINKED ISSUE #130: R8/Proguard rules don't match `2.2.0`'s code
I'm using mordant via `clikt` and its latest release `4.2.1` pulls the most recent mordant version which according to what I saw started providing their own Proguard/R8 rules (🚀). Unfortunately the rules were not updated to match latest code changes, and the build now fails with
```
Error: Missing class com.oracle.svm.core.annotate.Delete (referenced from: com.github.ajalt.mordant.internal.jna.JnaLinuxMppImpls and 5 other contexts)
Missing class org.graalvm.nativeimage.Platform$LINUX (referenced from: com.github.ajalt.mordant.internal.nativeimage.NativeImagePosixMppImpls and 1 other context)
Missing class org.graalvm.nativeimage.Platform$MACOS (referenced from: com.github.ajalt.mordant.internal.nativeimage.NativeImagePosixMppImpls and 1 other context)
Missing class org.graalvm.nativeimage.Platform$WINDOWS (referenced from: com.github.ajalt.mordant.internal.nativeimage.NativeImageWin32MppImpls and 1 other context)
Missing class org.graalvm.nativeimage.Platforms (referenced from: com.github.ajalt.mordant.internal.nativeimage.NativeImagePosixMppImpls and 3 other contexts)
Missing class org.graalvm.nativeimage.StackValue (referenced from: kotlin.Pair com.github.ajalt.mordant.internal.nativeimage.NativeImagePosixMppImpls.getTerminalSize() and 3 other contexts)
Missing class org.graalvm.nativeimage.c.CContext$Directives (referenced from: com.github.ajalt.mordant.internal.nativeimage.PosixLibC$Directives and 1 other context)
Missing class org.graalvm.nativeimage.c.CContext (referenced from: com.github.ajalt.mordant.internal.nativeimage.PosixLibC and 1 other context)
Missing class org.graalvm.nativeimage.c.constant.CConstant (referenced from: int com.github.ajalt.mordant.internal.nativeimage.PosixLibC.STDIN_FILENO() and 4 other contexts)
Missing class org.graalvm.nativeimage.c.function.CFunction (referenced from: int com.github.ajalt.mordant.internal.nativeimage.PosixLibC.ioctl(int, int, com.github.ajalt.mordant.internal.nativeimage.PosixLibC$winsize) and 4 other contexts)
Missing class org.graalvm.nativeimage.c.struct.CField (referenced from: short com.github.ajalt.mordant.internal.nativeimage.PosixLibC$winsize.getWs_col() and 5 other contexts)
Missing class org.graalvm.nativeimage.c.struct.CStruct (referenced from: com.github.ajalt.mordant.internal.nativeimage.PosixLibC$winsize and 1 other context)
Missing class org.graalvm.nativeimage.c.type.CIntPointer (referenced from: boolean com.github.ajalt.mordant.internal.nativeimage.NativeImageWin32MppImpls.stdinInteractive() and 2 other contexts)
Missing class org.graalvm.word.PointerBase (referenced from: kotlin.Pair com.github.ajalt.mordant.internal.nativeimage.NativeImagePosixMppImpls.getTerminalSize() and 8 other contexts)
Compilation failed
```

So I guess grallvm-related references should be kept somehow too? I'm not sure how they are used, so maybe ignoring them is the proper choice (`-dontwarn org.graalvm.**`)? 

If needed, failing build can be found here: https://github.com/usefulness/diffuse/actions/runs/6421384714/job/17435616654?pr=238
build scan: https://scans.gradle.com/s/fkzlsu76u35ce/console-log?anchor=162&page=1

### PR DESCRIPTION
**Pull request by ajalt**
Add a smoke test that builds and r8 fatjar and runs it to prevent future regressions.

Fixes #130

### CODE REVIEW

File: mordant/src/jvmMain/resources/META-INF/proguard/mordant.pro [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # Keep rules for those who are using ProGuard.
 
 -keep class com.sun.jna.** { *; }
--keep class * implements com.sun.jna.** { *; }
\ No newline at end of file
+-keep class * implements com.sun.jna.** { *; }
+-keepattributes *Annotation*
```
Comment (JakeWharton) [inline mordant/src/jvmMain/resources/META-INF/proguard/mordant.pro:5]: This is a very broad rule that keeps a lot of things that I suspect are not needed. Usually it's recommended to do something like `RuntimeVisibleAnnotations,RuntimeVisibleParameterAnnotations,RuntimeVisibleTypeAnnotations,AnnotationDefault` instead which allows the removal of runtime-invisible annotations. The full list of attributes is available here: https://www.guardsquare.com/manual/configuration/attributes

Comment (ajalt) [reply mordant/src/jvmMain/resources/META-INF/proguard/mordant.pro]: Awesome, thanks for the insight!

**Comment by JakeWharton** [comment]
Any chance we can get a new release with these? Or are you waiting for the progress bar rewrite to complete?

**Comment by ajalt** [comment]
I was, but since it's taking a while I just released 2.3.0 so you don't have to wait.

**Pull request merged**